components/local-app/pkg/auth/auth.go

```diff
@@ -58,13 +59,28 @@ const html = `
 	</body>
 </html>`
 
+// NOTE: the port ranges all need to be valid redirect URI's in the backend
+const STARTING_PORT_NUM = 63110
+const ENDING_PORT_NUM = 63120
+
 // Login walks through the login flow for obtaining a Gitpod token
 func Login(ctx context.Context, opts LoginOpts) (token string, err error) {
-	rl, err := net.Listen("tcp", "localhost:64110")
-	if err != nil {
-		return "", err
+	// Try a range of ports
+	var rl net.Listener
+	port := STARTING_PORT_NUM
+	for port < ENDING_PORT_NUM {
```

Comment:
A more classic `for` loop would be easier to understand here I reckon (no searching for `port++`).